### PR TITLE
ISPN-9223 1PC isn't properly implemented in Hot Rod Client

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.jboss.org/browse/ISPN-9223

### CODE REVIEW

**Comment by tristantarrant** [comment]
I triggered a new CI build as there were many failures: all of them unrelated, but better be safe

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/XaModeTransactionTable.java [hunk 1]
```diff
@@ -310,10 +316,10 @@ private void internalCommit() throws XAException {
          // commits only.
       }
 
-      private int internalPrepare(boolean onePhaseCommit) throws XAException {
+      private int internalPrepare() throws XAException {
          boolean readOnly = true;
          for (TransactionContext<?, ?> ctx : registeredCaches.values()) {
-            switch (ctx.prepareContext(currentXid, onePhaseCommit)) {
+            switch (ctx.prepareContext(currentXid, false)) {
```
Comment (wburns) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/XaModeTransactionTable.java]: It shouldn't hold this up, but it looks like we can just remove the `onePhaseCommit` argument to `TransactionContext.prepareCommit`?

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/XaModeTransactionTable.java [hunk 2]
```diff
@@ -333,6 +339,45 @@ private int internalPrepare(boolean onePhaseCommit) throws XAException {
...
+         outer: for (int i = 0; i < size - 1; ++i) {
+            TransactionContext<?,?> ctx = txCaches.get(i);
+            switch (ctx.prepareContext(currentXid, false)) {
+               case XAResource.XA_OK:
+                  preparedCaches.add(ctx);
+                  break;
+               case Integer.MIN_VALUE:
+                  //signals a marshaller error of key or value. the server wasn't contacted
+                  commit = false;
+                  break outer;
```
Comment (wburns) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/XaModeTransactionTable.java]: Seems like the break goto isn't needed anymore?

**Comment by wburns** [comment]
Integrated into master, thanks @pruivo !

**Pull request merged**